tests/test_generator.py

```diff
@@ -0,0 +1,89 @@
+from unittest.mock import MagicMock, patch, call
+
+from application import generators as GENERATORS
+
+
+@patch.object(GENERATORS, 'ascii_uppercase', 1)
+@patch.object(GENERATORS, 'ascii_lowercase', 2)
+def test_generate_random_image():
+
+    GENERATORS.randint = MagicMock(return_value=3)
+    GENERATORS.Image.new = MagicMock(return_value=4)
+    GENERATORS.choice = MagicMock(return_value=5)
+
+    result_of_the_call_GENERATORS_str_method = MagicMock()
+    GENERATORS.str = MagicMock(return_value=result_of_the_call_GENERATORS_str_method)
+    result_of_the_call_GENERATORS_str_method.join = MagicMock(return_value=6)
+
+    GENERATORS.ImageFont.truetype = MagicMock(return_value=7)
+
+    result_of_the_call_GENERATORS_ImageDraw_Draw_method = MagicMock()
+    GENERATORS.ImageDraw.Draw = MagicMock(return_value=result_of_the_call_GENERATORS_ImageDraw_Draw_method)
+    result_of_the_call_GENERATORS_ImageDraw_Draw_method.text = MagicMock()
+
+    counter = 0
+    for image in GENERATORS.generate_random_image():
+
+        assert image == 4
+
+        assert GENERATORS.randint.call_count == 10
+        expected_calls_for_GENERATORS_randint = [call(0, 255), call(0, 255), call(0, 255), call(100, 300),
+                                                 call(100, 300), call(1, 15), call(10, 20), call(0, 255),
+                                                 call(0, 255), call(0, 255)]
+        assert GENERATORS.randint.call_args_list == expected_calls_for_GENERATORS_randint
+        GENERATORS.randint.reset_mock()
+
+        GENERATORS.Image.new.assert_called_once_with('RGB', (3, 3), color=(3, 3, 3))
+        GENERATORS.Image.new.reset_mock()
+
+        assert GENERATORS.choice.call_count == 3
```

Comment:
W przypadku funkcji z yield za każdym razem resetuję Mocki, aby sprawdzić czy za każdym razem dana funkcja była wywołana określoną liczbę razy. Czy tak to może zostać czy nie resteować Mocków i sprawdzić liczbę wywołań po wyjściu z pętli ?